projects/laji/src/app/shared/service/dialog.service.ts

```diff
@@ -17,31 +42,33 @@ export class DialogService {
     if (this.platformService.isServer) {
       return ObservableOf(onServer);
     }
-    return this.createDialog(message, false);
+    return this.createDialog<AlertConfig, boolean>({message, showCancel: false});
   }
 
-  confirm(message: string, onServer = false): Observable<boolean> {
+  confirm(message: string, confirmLabel?: string, onServer = false): Observable<boolean> {
     if (this.platformService.isServer) {
       return ObservableOf(onServer);
     }
-    return this.createDialog(message);
+    return this.createDialog<ConfirmConfig, boolean>({message, showCancel: true, confirmLabel});
   }
 
-  prompt(message: string, _default?: string): Observable<string | null> {
-    return this.createDialog(message, true, true, _default);
+  prompt(message: string, confirmLabel?: string, _default?: string): Observable<string | null> {
+    return this.createDialog<PromptConfig, string | null>({message, showCancel: true, confirmLabel, prompt: true, promptValue: _default});
   }
 
-  private createDialog(message: string, showCancel?: boolean): Observable<boolean>;
-  private createDialog(message: string, showCancel: boolean, prompt: true, promptDefault?: string): Observable<string | null>;
-  private createDialog(message: string, showCancel = true, prompt = false, promptDefault?: string): Observable<boolean | string | null> {
+  private createDialog<T extends DialogConfig, R = (boolean | (string | null))>(options: T): Observable<R> {
+    const initialState: InitialState = {showCancel: true, prompt: false,  promptValue: '', ...options};
+    if (typeof options.confirmLabel === 'string') {
+      initialState.confirmLabel = options.confirmLabel;
```

Comment:
This doesn't seem to do anything since `initialState` is set to `{showCancel: true, prompt: false,  promptValue: '', ...options}`

Reply:
Good catch :+1:  I'll remove it